.github/workflows/test.yml

```diff
@@ -0,0 +1,55 @@
+name: "Test"
+
+on:
+    push:
+        branches: [master]
+    pull_request:
+        branches: [master]
+
+jobs:
+    unit-tests-linux:
+        name: Tests (Linux)
+        runs-on: ubuntu-latest
+        permissions:
+            actions: read
+            contents: read
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v3
+            - name: Setup .NET Core
+              uses: actions/setup-dotnet@v4
+              with:
+                  dotnet-version: |
+                      2.1.x
+                      3.0.x
+                      3.1.x
+                      5.0.x
+                      6.0.x
+                      7.0.x
+                      8.0.x
+                      9.0.x
+            - name: Run tests
+              run: dotnet test --blame-hang --blame-hang-dump-type none --blame-hang-timeout 60s -- tests/StatsdClient.Tests/
+    unit-tests-windows:
+        name: Tests (Windows)
+        runs-on: windows-latest
+        permissions:
+            actions: read
+            contents: read
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v3
+            - name: Setup .NET Core
+              uses: actions/setup-dotnet@v4
+              with:
+                  # Windows runner images have more versions of .NET Core pre-installed than Linux.
+                  dotnet-version: |
+                      2.1.x
+                      3.0.x
+                      3.1.x
+                      5.0.x
+                      7.0.x
+            - name: Run tests
+              run: dotnet test --blame-hang --blame-hang-dump-type none --blame-hang-timeout 60s -- tests/StatsdClient.Tests/
```

Comment:
Small NIT, we have a build script `BuildWarningAsError.ps1` we might want to also run as part of CI or at least the same commands it calls.

Reply:
Oooh, nice. I'll take a look at that as a follow-up PR.